### PR TITLE
[FEAT] Get Plans of History

### LINKED ISSUE #35: [FEAT] Plan History의 Plan 조회 Acceptance Test
# Contents

Please briefly explain the issue

## Description

조회 요청은 다음과 같다: GET api/goal/{goalId}/plan-histories/{planHistoryId}
반환은 page 형식으로 제공된다.
해당 조회는 다음과 같은 시나리오에서 사용된다.
- 계획 피드백 완료 시
- 계획 기록 상세페이지

## Todo

-   [ ] GoalPlanHistoryController getPlans
-   [ ] GoalPlanHistoryAcceptanceTest getPlans

## Check Priority

Expected Priority:

## Reference (Optional)

(null)

### LINKED ISSUE #36: [FEAT] Plan History의 Plan 리스트 조회 Service
# Contents

Please briefly explain the issue

## Description

조회는 aggregate root를 거치지 않아도 된다. -> PlanHistoryService
단, 가져오는 것은 Root를 통해서 가져오는 것처럼 구현한다. -> GoalRepository

## Todo

-   [ ] PlanHistoryService getPlans
-   [ ] 단위 테스트
-   [ ] GoalRepository getPlansByPlanHistory

## Check Priority

Expected Priority:

## Reference (Optional)

(null)

### PR DESCRIPTION
**Pull request by sueelly**
## #️⃣ Related Issue

Linked Feature backlog: close #35 close #36 

## 📝 Purpose of PR

히스토리 내의 완료 계획들을 조회한다.

## Contents

- goal / history가 삭제되었을 경우 forbiddenexception

### Issue: QueryService?
실은 query로 가져오는 것은 controller에서 바로 가져와도 상관이 없다. 로직이랄 게 없으니.
다만 여기서 goal, history 등이 deleted 되었을 경우 예외를 던진다는 로직이 존재하기 때문에 따로 서비스를 두었다.

### Issue: 어디 패키지에 속하는 게 옳을까
request url은 goal, history 둘 다 갖고
plan history에도 해당하며, 결국 조회하는 것은 plan이다.
planQuery로 두는 것은 이견이 없어 planQuerySerivce에 두었다.
추구 planQueryRepository가 분리되면 (QueryDSL과 함께) 그때는 나누는 게 유의미해질 듯 하다.

### Issue: QueryDSL

쿼리가 복잡해지고, 쿼리에 대해 컴파일 시에 에러를 잡기 위해 queryDSL 도입을 시도했다.
하지만 의존성 처리 과정에서 에러가 발생하여 build 자체가 되지 않아 철회함
추후 추가해보자

## Checklist

-   [x] Does commit messages follow the convention?
-   [x] Are variable names brief but descriptive?
-   [x] Is the code clean and easy to understand?
-   [x] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [ ] Does the test successfully test edge and corner cases?

## 💬 Review Requirements (Optional)

Is there anything you can praise about this PR? Start with praise.
Make a review, leaving kind comments and suggesting changes where needed (to resolve the above).

### CODE REVIEW

File: src/main/java/ac/dnd/dodal/common/response/ApiResponse.java [hunk 1]
```diff
@@ -58,10 +58,23 @@ public static <T> ApiResponse<T> failure(ResultCode resultCode) {
...
     public static <T> ApiResponse<T> failure(ResultCode resultCode, String message) {
-        return new ApiResponse<>(
-            resultCode.getCode(),
-            message,
-            null
-        );
+        return new ApiResponse<>(resultCode.getCode(), message, null);
+    }
+
+    public record PageResponse<T>(
```
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/common/response/ApiResponse.java:64]: 개인적인 궁금증이긴 한데, Page를 도입해서 얻을 수 있는 장점은 어떤 것들이 있나요? 페이징을 해보진 않아서 고민해보지 못했던 부분이라 궁금해요

File: src/test/java/ac/dnd/dodal/acceptance/goal/steps/GoalSteps.java [hunk 2]
```diff
@@ -7,10 +7,9 @@
 import io.restassured.response.Response;
 import io.restassured.http.ContentType;
 
-import ac.dnd.dodal.AcceptanceTest;
 import ac.dnd.dodal.ui.goal.request.CreateGoalRequest;
 
-public class GoalSteps extends AcceptanceTest {
+public class GoalSteps {
```
Comment (5hseok) [inline src/test/java/ac/dnd/dodal/acceptance/goal/steps/GoalSteps.java:12]: AcceptanceTest의 설정들을 사용하지 않아서 상속 관계를 삭제한건가요?

**Review by 5hseok — APPROVED** [review]

File: src/test/java/ac/dnd/dodal/acceptance/goal/steps/GoalSteps.java (hunk 2 above)
Comment (sueelly) [reply src/test/java/ac/dnd/dodal/acceptance/goal/steps/GoalSteps.java:12]: 넵 그렇습니다

File: src/main/java/ac/dnd/dodal/common/response/ApiResponse.java (hunk 1 above)
Comment (sueelly) [reply src/main/java/ac/dnd/dodal/common/response/ApiResponse.java:64]: 페이징의 목적을 생각해보시면 될 것 같아요.
결국 데이터를 한번에 모두 가져올 수는 없으니, 쪼개서 가져와야하는데
몇번째 데이터부터(page/number), 몇개씩(size) 가져올지 정해서 가져오는 거라고 보시면 돼요.
예를 들어 하나의 목표에 계획이 1000개가 있다고 할 때 한번에 모두 반환할 수는 없으니까요
